web3_proxy/src/frontend/users/subuser.rs

```diff
@@ -208,7 +208,7 @@ pub async fn modify_subuser(
 
     debug!("User tier is: {:?}", user_tier);
     // TODO: This shouldn't be hardcoded. Also, it should be an enum, not sth like this ...
-    if user_tier.id != 6 {
+    if UserTier::try_from(user_tier.title.as_str())? != UserTier::Premium {
```

Comment:
Don't require premium to create a referral. Only require it to actually get credits for referrals

Reply:
I will open a new issue for this